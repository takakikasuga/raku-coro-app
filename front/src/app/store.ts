import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import userReducer from '../features/user/userSlice';
import threadReducer from '../features/thread/threadSlice';
import dailyInfectionRuducer from '../features/graphs/dailyInfectionSlice';
import graph2Ruducer from '../features/graphs/graph2Slice';
import prefectureRuducer from '../features/common/prefectureSlice';
import dailyDeadRuducer from '../features/graphs/dailyDeadSlice';
import dailyCoronaRuducer from '../features/graphs/dailyCoronaSlice';
import totalCoronaRuducer from '../features/graphs/totalCoronaSlice';
import totalRecoveryRuducer from '../features/graphs/totalRecoverySlice';
import totalDethRuducer from '../features/graphs/totalDethSlice';
import vaccinationRuducer from '../features/graphs/vaccinationSlice';
import bedOccupancyRateRuducer from '../features/graphs/bedOccupancyRateSlice';
import dailyDethRuducer from '../features/graphs/dailyDethSlice';
import dailyRecoveryRuducer from '../features/graphs/dailyRecoverySlice';
import dailyPositiveRuducer from '../features/graphs/dailyPositiveSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    user: userReducer,
    thread: threadReducer,
    dailyInfection: dailyInfectionRuducer,
    graph2: graph2Ruducer,
    prefecture: prefectureRuducer,
    dailyDead: dailyDeadRuducer,
    dailyCorona: dailyCoronaRuducer,
    totalCorona: totalCoronaRuducer,
    totalRecovery: totalRecoveryRuducer,
    totalDeth: totalDethRuducer,
    vaccination: vaccinationRuducer,
    bedOccupancyRate: bedOccupancyRateRuducer,
    dailyDeth: dailyDethRuducer,
    dailyRecovery: dailyRecoveryRuducer,
    dailyPositive: dailyPositiveRuducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
