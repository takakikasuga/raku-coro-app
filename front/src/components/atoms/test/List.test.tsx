import React from 'react';
import {
  render,
  cleanup,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react';
import { afterEach, expect } from '@jest/globals';
import { store } from '../../../app/store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { List } from '../index';

interface ListPropsInterface {
  children: string;
  url: string;
}

const ListProps: ListPropsInterface = {
  children: 'テスト',
  url: 'https://www.mhlw.go.jp/stf/covid-19/open-data.html',
};

describe('Footerコンポーネント', () => {
  // 各テストの直前に実行される（コンポーネントの取得）
  beforeEach(() => {
    render(
      <Provider store={store}>
        <List children={ListProps.children} url={ListProps.url} />
      </Provider>
    );
  });

  // 各処理が通った後にアンマウントさせる
  afterEach(() => cleanup());

  describe('jest renderer test', () => {
    test('render correctly', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <List children={ListProps.children} url={ListProps.url} />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('aタグテスト', () => {
    test('aタグにPropsで渡した値が存在するか否か', () => {
      expect(screen.getByRole('link').textContent).toBe('テスト');
    });
    test('aタグにtarget属性のテスト', () => {
      expect(screen.getByRole('link')).toHaveAttribute('target', '_blank');
    });
    test('aタグにrel属性のテスト', () => {
      expect(screen.getByRole('link')).toHaveAttribute(
        'rel',
        'noopener noreferrer'
      );
    });
    test('aタグにhref属性のテスト', () => {
      expect(screen.getByRole('link')).toHaveAttribute(
        'href',
        'https://www.mhlw.go.jp/stf/covid-19/open-data.html'
      );
    });
  });
});
