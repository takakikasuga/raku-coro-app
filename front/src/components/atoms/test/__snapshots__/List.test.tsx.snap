// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footerコンポーネント jest renderer test render correctly 1`] = `
<li>
  <a
    href="https://www.mhlw.go.jp/stf/covid-19/open-data.html"
    rel="noopener noreferrer"
    target="_blank"
  >
    テスト
  </a>
</li>
`;
